spikeinterface/preprocessing/phase_shift.py

```diff
@@ -47,17 +47,19 @@ def __init__(self, recording, margin_ms=40.,  inter_sample_shift=None, dtype=Non
         
         margin = int(margin_ms * recording.get_sampling_frequency() / 1000.)
         
-        force_dtype_back = None
         if dtype is None:
-            if recording.get_dtype().kind in ('i', 'u'):
-                # because of the tapper on margin we need to force the computation to float32
-                # and then go back to the original buffer unless dtype is explicitly forced
-                dtype = np.dtype('float32')
-                force_dtype_back = recording.get_dtype()
- 
+            dtype = recording.get_dtype()
+        # the "apply_shift" function returns a float64 buffer. In case the dtype is different
+        # than float64, we need a temporary casting and to force the buffer back to the original dtype
+        if dtype.str != "<f8":
```

Comment:
```suggestion
        if dtype != "float64":
```

Reply:
Here you were forcing the endianness.

Reply:
done (used `str(dtype)` because dtype could be a np object)